hpx/lcos/promise.hpp

```diff
@@ -139,20 +133,19 @@ namespace lcos {
         //   - promise_already_satisfied if its shared state already has a
         //     stored value or exception.
         //   - no_state if *this has no shared state.
-        void set_exception(
-            boost::exception_ptr const& e, error_code& ec = throws)
-        {
-            base_type::set_exception(e, ec);
-        }
+        using base_type::set_exception;
```

Comment:
I should probably revert those changes. We have not done it this way (`using base_type::...`) in other places either.

Also, if we decided to stick with it, we could easily use doxygens `\copydoc` directive to pull in the docs from the base class, if needed.